### ISSUE TITLE
Questions of Model Compression: Pruning & Quantization

### ISSUE THREAD

**Issue by MT010104** [microsoft/nni#5305]
Thanks for the great projects. I have some questions while using NNI.
1. Quantization doesn't need ModelSpeedup, right?
2. Finetuning is necessary for pruning and quantization?
3. Models are bigger after quantization. I got some tips in other issues, but I couldn't make it out. Is this phenomenon common? How is the degree of quantization reflected?

**Comment by QuanluZhang** [comment]
For the first question, not really, quantization also needs ModelSpeedup, because quantization algorithms use simulated quantization (i.e., quant and dequant).

For the second question, finetuning is not a must, but the pruning and quantization that are based on finetuning give you higher model accuracy.

For the third question, the model becomes bigger because of the simulated quantization. ModelSpeedup is required to obtain the real acceleration. You can refer to this pr (#5039 ) for details about how to speed up a quantized model.

**Comment by MT010104** [comment]
Thanks for the rely. ModelSpeedup after quantization needs to install tensorrt and pycuda. Is pycuda necessary? Because there isn't suitable pycuda for my environment.

**Comment by QuanluZhang** [comment]
> Thanks for the reply. ModelSpeedup after quantization needs to install tensorrt and pycuda. Is pycuda necessary? Models after Speedup aren't normal PyTorch models and they need specific functions such as "inference()"?

yes, pycuda is required because tensorrt's python api replies on pycuda.
yes, they need specific functions, e.g., "inference()".

**Comment by QuanluZhang** [comment]
feel free to reopen if you have any other question.
